### ISSUE TITLE
aws_cloudwatch_metric_alarm AlarmActions ARN incorrectly rejected in 1.41.0

### ISSUE THREAD

**Issue by jeohist** [terraform-providers/terraform-provider-aws#6203]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.11.8

### Affected Resource(s)

* aws_cloudwatch_metric_alarm

### Terraform Configuration Files

```hcl
data "aws_region" "current_region" {}

resource "aws_cloudwatch_metric_alarm" "foo" {
  count               = 1
  alarm_name          = "foo"
  alarm_description   = "Status checks have failed for system, recovering instance"
  namespace           = "AWS/EC2"
  statistic           = "Minimum"
  comparison_operator = "GreaterThanThreshold"
  threshold           = "0"
  unit                = "Count"
  evaluation_periods  = "2"
  period              = "60"
  metric_name         = "StatusCheckFailed_System"
  alarm_actions       = ["arn:aws:automate:${data.aws_region.current_region.name}:ec2:recover"]

  dimensions {
    InstanceId = "${var.instanceid}"
  }
}
```

### Expected Behavior

aws_cloudwatch_metric_alarm.foo should have been created

### Actual Behavior

* resource.aws_cloudwatch_metric_alarm.foo: "alarm_actions.0" doesn't look like a valid ARN ("^arn:[\\w-]+:([a-zA-Z0-9\\-])+:([a-z]{2}-(gov-)?[a-z]+-\\d{1})?:(\\d{12})?:(.*)$"): "arn:aws:automate:eu-west-1:ec2:recover"

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

This broke since #6151 was merged

### References

* AWS Documentation: https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricAlarm.html
* #6151

**Comment by bflad** [comment]
Sorry for the trouble! Bug fix pull request submitted with acceptance testing to cover these specially formatted ARNs: #6206

**Comment by jeohist** [comment]
@bflad No problem! Appreciate the quick fix :-)

**Comment by Seraf** [comment]
Hello,

Thanks for the quick fix ! Is a hotfix release planned in order to quickly fix this bug upstream ?

**Comment by bflad** [comment]
This will likely be released on the normal weekly cadence. As a workaround, you can pin your configuration to the previous provider release, e.g.

```hcl
provider "aws" {
  # ... potentially other configuration ...
  version = "1.40.0"
}
```

**Comment by prasanthsurapaneni** [comment]
Thank you.

**Comment by prasanthsurapaneni** [comment]
> This will likely be released on the normal weekly cadence. As a workaround, you can pin your configuration to the previous provider release, e.g.
> 
> ```terraform
> provider "aws" {
>   # ... potentially other configuration ...
>   version = "1.40.0"
> }
> ```

This does not seem to work. changed the root provider version and all the other to "1.40" still persisting.

**Comment by bflad** [comment]
Thanks for your patience -- the fix for this has been merged and will release with version 1.42.0 of the AWS provider, likely today or tomorrow.

**Comment by swoodford** [comment]
I am also having fun troubleshooting this bug today, trying to run a `terraform plan` against an existing infrastructure with a CloudWatch `StatusCheckFailed` alarm that was previously created and working fine in a `terraform apply`:

`Error: aws_cloudwatch_metric_alarm.status-check: "alarm_actions.0" doesn't look like a valid ARN ("^arn:[\\w-]+:([a-zA-Z0-9\\-])+:([a-z]{2}-(gov-)?[a-z]+-\\d{1})?:(\\d{12})?:(.*)$"): "arn:aws:automate:us-east-1:ec2:recover"`
